src/redux/deviceRedux.tsx

```diff
@@ -0,0 +1,27 @@
+import { Store } from "./initialState";
+import { ActionType } from "./store";
+
+/* selectors */
+export const getDevice = (isDesktop: Store) => isDesktop;
```

Comment:
Oof, thats kinda unfortunate. 

First of all, `isDesktop` is a misleading variable name.

More importantly, you are returning the whole store from the selector, meaning any action dispatch will cause store object to be recreated, and every place that depended on the `isDesktop` check will be re-run (basically re-rendering a considerable parts of the react tree when it doesn't need to).

Selectors are needed for the purpose of limiting the effects of state updates only to the parts that depend on the changed value. In order to prevent redundant updates redux (and notably react itself) runs an equality check _([kinda](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is))_ and skips re-renders if the value is the same (I think redux allows you to customise comparison function in `useSelector`)

Since you've removed `device` prop, I'd advise you to do something like that:
```suggestion
export const isDesktop = (store: Store) => store.isDesktop;
```
and changing all of the places you depend on `getDevice`.

This will ensure that those places re-run only when `isDesktop` changes from true to false or vice-versa.